compiler/hash-ast/src/keyword.rs

```diff
@@ -0,0 +1,65 @@
+//! Hash Compiler language keyword definitions
+//!
+//! All rights reserved 2021 (c) The Hash Language authors
+
+use std::fmt;
+use std::slice::Iter;
+
+pub enum Keyword {
+    Let,
+    For,
+    While,
+    Loop,
+    If,
+    Else,
+    Match,
+    As,
+    In,
+    Where,
+    Trait,
+    Enum,
+    Struct,
+    Continue,
+    Break,
+    Return,
+}
+
+impl Keyword {
+    pub fn iterator() -> Iter<'static, Keyword> {
+        #[rustfmt::skip]
+        static DIRECTIONS: [Keyword; 16] = [
+            Keyword::Let, Keyword::For, Keyword::While, Keyword::Loop,
+            Keyword::If, Keyword::Else, Keyword::Match, Keyword::As,
+            Keyword::In, Keyword::Where, Keyword::Trait, Keyword::Enum,
+            Keyword::Struct, Keyword::Continue, Keyword::Break, Keyword::Return,
+        ];
+        DIRECTIONS.iter()
```

Comment:
There won't be any error if a new keyword gets added to the enum but not this list. There's a crate https://crates.io/crates/strum which might be useful here, which would autogenerate this method (and to_str for that matter). What do you think about using that?

Reply:
I was looking for something like that, but I couldn't find anything particularly good. I'll check it out and have a look at it. Also, is there a way to create the hashmap at compile time to insert the keywords?

Reply:
Unfortunately not. The map has to be created at runtime. However, it can be written as a lazy static. (See `lazy_static` crate).